subprojects/core/src/main/java/org/gradle/internal/model/CalculatedValueContainerFactory.java

```diff
@@ -46,27 +47,21 @@ public <T, S extends ValueCalculator<? extends T>> CalculatedValueContainer<T, S
         return new CalculatedValueContainer<>(displayName, supplier, projectLeaseRegistry, globalContext);
     }
 
-    /**
-     * A convenience to create a calculated value that has no dependencies and that does not access any mutable model state.
-     */
-    public <T> CalculatedValueContainer<T, ?> create(DisplayName displayName, Supplier<? extends T> supplier) {
+    @Override
+    public <T> CalculatedValueContainer<T, ValueCalculator<T>> create(DisplayName displayName, Supplier<? extends T> supplier) {
```

Comment:
> I'm not suggesting that but, an alternative would be to introduce a covariant generic parameter to stand in for `ProjectInternal`.

It might be possible to introduce another abstraction to wrap the Project, but I'd leave this for maintenance day.